### PR TITLE
fix: WooPay 'Save my info' spacing

### PR DESCRIPTION
**Pull request by frosso**
#### Changes proposed in this Pull Request

I noticed that WooPay's "Save my info" addins unnecessary spacing (via margins) to the checkout pages when no content is displayed.
As good citizens of the checkout pages, fixing to ensure consistent behavior compared to when WooPay is disabled or not available.

Cannot be tested via UTs.

Before/After - WooPay disabled - spacing is consistent:
Screenshot 2023-10-06 at 12 19 30 PM

Before - no payment method needed - "Save my info" is adding a margin-top:
Screenshot 2023-10-06 at 12 24 47 PM

Before - non-CC payment method is selected (or existing WooPay user is being used) - "Save my info" is adding a margin-top:
Screenshot 2023-10-04 at 3 41 03 PM

After - non-CC payment method is selected - "Save my info" is no longer adding a margin-top:
Screenshot 2023-10-04 at 3 39 07 PM

Before/After - CC Payment method is selected:
Screenshot 2023-10-04 at 3 39 19 PM

#### Testing instructions

- Enable WooPay in the WC Payments settings
- Enable other non-WooPayments payment methods (e.g.: cash on delivery)
- Add a simple product to the cart, go to checkout
- Select CC as a payment method
- Enter a valid WooPay email
  - "Save my info" no longer displayed, notice spacing above "pay now" and below the payment methods
- Enter a non-existing WooPay email
  - "Save my info" is displayed, notice the spacing
- Select a non-CC payment method (e.g.: cash on delivery)
  - "Save my info" no longer displayed and no longer adding a margin-top

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/settings/phone-input/style.scss [hunk 1]
```diff
@@ -6,7 +6,21 @@
 	display: flex;
 	flex-direction: column;
 	gap: $gap;
-	margin-top: $gap-large;
```
Comment (frosso) [inline client/settings/phone-input/style.scss:9]: This margin was applied even when the "save my info" container didn't have any content.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7405` or branch name `fix/woopay-save-my-info-styles` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/woopay-save-my-info-styles) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7bf94f7ab1d048fb773a36914ab914db2e9773c2
- Build time: 2023-10-13 07:22:55 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/phone-input/style.scss [hunk 2]
```diff
@@ -106,10 +115,6 @@
 	.save-details-form.place-order {
 		margin-top: $gap-smaller;
 	}
-
-	.woopay-save-new-user-container + .place-order {
-		margin-top: $gap-large !important;
```
Comment (frosso) [inline client/settings/phone-input/style.scss:111]: Removing uncommented `!important` is always good.

**Review by mdmoore — APPROVED** [review]
LGTM! 🚢

**Pull request merged**